pkg-manager/resolve-dependencies/src/index.ts

```diff
@@ -334,16 +415,20 @@ async function finishLockfileUpdates (
         // Optional dependencies are not always downloaded, so there is no way to know whether they need to be built or not.
         requiresBuild = true
       } else {
+        if (useExperimentalNpmjsFilesIndex) {
```

Comment:
This should also be only true when lockfileOnly=true

Reply:
i think it already is, with the change to set opts.LockfileOnly && opts.useExperimentalNpmjsFilesIndex